### DISCUSSION TITLE
How to add a seperate layout for the root page

### DISCUSSION THREAD

**Discussion by odincreek** [vercel/next.js#46847] [Help]
### Summary

So I'm using NextJS 13.2 and in my src/app folder I have a layout.js file, but there is also the home page the src/app/page.js file.

I want to have a gray background on my <body> for the home page and some other pages in the root folder, but I don't want to put it in the root layout because it also sends the background color to my dashboard page, and I don't want to put it in each page wrapped with a div.

What is the best way to do this, I tried making a (home) folder with page.js and layout.js, but it still gets the root layout.

How can I solve so that my home page, about page etc in the root folder gets a gray background and specific next/font from a layout, but items in the dashboard folder should not get that gray background or font.

**Comment by wwwdepot** [comment]
Have you tried using multiple groups?

```
/app/(home)/layout.js
/app/(dashboard)/dashboard/layout.js
```

**Comment by guitheengineer** [comment]
You can accomplish different root layouts using [route groups](https://beta.nextjs.org/docs/routing/defining-routes#route-groups)

![image](https://user-images.githubusercontent.com/49620737/213922519-a5fec86a-f845-439b-90c2-1f5b8a6b789d.png)


In your case it would work by putting home, blog and faq in a group with a root layout, then dashboard and profile on another group with a different layout.js.

**Comment by odincreek** [comment]
The above answer did clear out some questions however with this approach I'd have to put for example my custom font using next/font/local again in there and the className for the html and body tags. Feels wrong.

Is there a possibility to create a root layout inside of the app folder so it would be:
app/layout.js

And inside of this file at app/layout.js add a bg-gray for example to the body and some other class to the html tag, and then in the dashboard group or some other route I'd like to use the same layout just with different class names in the html and body tags,

But I'd like to keep my React Query wrapper and my next/font/local without re-adding the same code in the dashboard layout or some other layout, so only the classname would change, currently when I'm trying some stuff I get hydration error, and if I leave out for example a className in the html or body I also get hydration error, really confusing :(

Thanks for the replies and assistance guys!

**Reply by irfansaeedkhan** [reply]
yes keep the app/layout.js and just add font family and global stuff that you want through out the app, it will be applied to all, then you can make group and do other changes

**Comment by irfansaeedkhan** [comment]
I have one question, what if i want specific layout inside sub groups and i dont want to take parent layout, is there workaround for that

**Reply by sabbir2609** [reply]
all sub will go through the `app/layout`
